From 2ff2c5fec2e5f0e65aa64fd9feecc2531d732ff0 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@strace.io>
Date: Wed, 17 Dec 2025 08:00:00 +0000
Subject: [PATCH 7/8] xlat: update BPF_* constants

* src/xlat/bpf_file_flags.in (BPF_F_RB_OVERWRITE): New constant
introduced by Linux kernel commit v6.19-rc1~171^2~73.
* src/xlat/bpf_map_types.in (BPF_MAP_TYPE_ARENA): New constant
introduced by Linux kernel commit v6.19-rc1~171^2~52^2~11.
* tests/bpf.c: Check it.
* NEWS: Mention this change.
---
 NEWS                       |  2 +-
 src/xlat/bpf_file_flags.in |  1 +
 src/xlat/bpf_map_types.in  |  1 +
 tests/bpf.c                | 10 +++++-----
 4 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/NEWS b/NEWS
index 9ea8a48ad..7f5393c7d 100644
--- a/NEWS
+++ b/NEWS
@@ -3,7 +3,7 @@ Noteworthy changes in release ?.?? (????-??-??)
 
 * Improvements
   * Updated decoding of struct mnt_id_req and struct perf_event_attr.
-  * Updated the list of IORING_OP_* constants.
+  * Updated lists of BPF_* and IORING_OP_* constants.
 
 Noteworthy changes in release 6.18 (2025-12-07)
 ===============================================
diff --git a/src/xlat/bpf_file_flags.in b/src/xlat/bpf_file_flags.in
index 1a280f875..df087ec99 100644
--- a/src/xlat/bpf_file_flags.in
+++ b/src/xlat/bpf_file_flags.in
@@ -7,3 +7,4 @@ BPF_F_VTYPE_BTF_OBJ_FD	(1U << 15)
 BPF_F_TOKEN_FD		(1U << 16)
 BPF_F_SEGV_ON_FAULT	(1U << 17)
 BPF_F_NO_USER_CONV	(1U << 18)
+BPF_F_RB_OVERWRITE	(1U << 19)
diff --git a/src/xlat/bpf_map_types.in b/src/xlat/bpf_map_types.in
index 682646801..eb3453769 100644
--- a/src/xlat/bpf_map_types.in
+++ b/src/xlat/bpf_map_types.in
@@ -35,3 +35,4 @@ BPF_MAP_TYPE_BLOOM_FILTER 30
 BPF_MAP_TYPE_USER_RINGBUF 31
 BPF_MAP_TYPE_CGRP_STORAGE 32
 BPF_MAP_TYPE_ARENA 33
+BPF_MAP_TYPE_INSN_ARRAY	34
diff --git a/tests/bpf.c b/tests/bpf.c
index 4ae83886d..d1b9bc261 100644
--- a/tests/bpf.c
+++ b/tests/bpf.c
@@ -279,7 +279,7 @@ init_BPF_MAP_CREATE_attr7(struct bpf_attr_check *check, size_t idx)
 	attr->map_ifindex = ifindex_lo();
 }
 
-static_assert(ARRAY_SIZE(bpf_map_types_xdata) == 34,
+static_assert(ARRAY_SIZE(bpf_map_types_xdata) == 35,
 	      "The map_type for tests 1 and 2 below needs to be updated");
 static struct bpf_attr_check BPF_MAP_CREATE_checks[] = {
 	{
@@ -290,7 +290,7 @@ static struct bpf_attr_check BPF_MAP_CREATE_checks[] = {
 	},
 	{ /* 1 */
 		.data = { .BPF_MAP_CREATE_data = {
-			.map_type = 33,
+			.map_type = 34,
 			.key_size = 4,
 			.value_size = 8,
 			.max_entries = 256,
@@ -300,7 +300,7 @@ static struct bpf_attr_check BPF_MAP_CREATE_checks[] = {
 			.map_name = "0123456789abcde",
 		} },
 		.size = offsetof(struct BPF_MAP_CREATE_struct, map_name) + 8,
-		.str = "map_type=BPF_MAP_TYPE_ARENA, key_size=4"
+		.str = "map_type=BPF_MAP_TYPE_INSN_ARRAY, key_size=4"
 		       ", value_size=8, max_entries=256"
 		       ", map_flags=BPF_F_NO_PREALLOC|BPF_F_NO_COMMON_LRU"
 				   "|BPF_F_NUMA_NODE|BPF_F_RDONLY|BPF_F_WRONLY"
@@ -312,7 +312,7 @@ static struct bpf_attr_check BPF_MAP_CREATE_checks[] = {
 	},
 	{ /* 2 */
 		.data = { .BPF_MAP_CREATE_data = {
-			.map_type = 34,
+			.map_type = 35,
 			.key_size = 0xface1e55,
 			.value_size = 0xbadc0ded,
 			.max_entries = 0xbeefcafe,
@@ -323,7 +323,7 @@ static struct bpf_attr_check BPF_MAP_CREATE_checks[] = {
 			.map_ifindex = 3141592653,
 		} },
 		.size = offsetofend(struct BPF_MAP_CREATE_struct, map_ifindex),
-		.str = "map_type=0x22 /* BPF_MAP_TYPE_??? */"
+		.str = "map_type=0x23 /* BPF_MAP_TYPE_??? */"
 		       ", key_size=4207812181, value_size=3134983661"
 		       ", max_entries=3203386110"
 		       ", map_flags=0xffffc000 /* BPF_F_??? */"
-- 
2.34.1

